client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/OutboundLineEditTable.tsx

```diff
@@ -70,15 +70,23 @@ const TotalRow = ({ allocatedQuantity }: { allocatedQuantity: number }) => {
   const formattedValue = useFormatNumber().round(allocatedQuantity, 2);
   const { data: prefs } = usePreference(
     PreferenceKey.SortByVvmStatusThenExpiry,
-    PreferenceKey.ManageVvmStatusForStock
+    PreferenceKey.ManageVvmStatusForStock,
+    PreferenceKey.AllowTrackingOfStockByDonor
   );
-  const hasExtraVVMColumn =
-    prefs?.manageVvmStatusForStock || prefs?.sortByVvmStatusThenExpiry;
+
+  const getColSpanToTotal = () => {
+    let colSpan = 5;
+    if (prefs?.allowTrackingOfStockByDonor) colSpan += 1;
+    if (prefs?.manageVvmStatusForStock || prefs?.sortByVvmStatusThenExpiry)
+      colSpan += 1;
+
+    return colSpan;
+  };
 
   return (
     <tr>
       <TotalCell colSpan={3}>{t('label.total-quantity')}</TotalCell>
-      <TotalCell colSpan={hasExtraVVMColumn ? 6 : 5}></TotalCell>
+      <TotalCell colSpan={getColSpanToTotal()}></TotalCell>
```

Comment:
Maybe there is a better way, to put the total cell in the right column based on which columns you have on?